gcc/testsuite/rust/compile/builtin_macro_recurse.rs

```diff
@@ -15,7 +15,7 @@ macro_rules! b {
 
 fn main() {
     // { dg-final { scan-tree-dump-times {"test1canary"} 1 gimple } }
-    let _ = concat!(a!(), 1, b!());
+    let _v = concat!(a!(), 1, b!());
```

Comment:
Why was this causing an issue?

Reply:
Something about not storing the string reference in a variable.

Reply:
I'm not getting any issue when running the file or another code sample like
```rust
fn main() {
    let _ = "a";
}
```

can you reproduce the behavior? I'd rather not change the test if everything is good, but at the same time I want to make sure this PR does not accidentally introduce a regression. The code above is valid Rust and so is the testcase

Reply:
If this
```
FAIL: rust/compile/builtin_macro_recurse.rs scan-tree-dump-times gimple ""test1canary"" 1
FAIL: rust/compile/closure_multi_args.rs (test for excess errors)
```
is the error you mean then there is an issue in your PR and we're now miscompiling wildcard patterns. Sorry :(